### PR TITLE
Improving error handling for invalid image inputs

### PR DESCRIPTION
**Pull request by lumoslnt**
This pull request includes changes to improve test coverage and handle empty image inputs in the `multimedia_utils` module. The most important changes include adding test cases for the `load_multimedia_data` and `create_image` functions, and modifying the exception handling for empty image inputs.

Summary of changes:

* <a href="diffhunk://#diff-032d36cff1825e9647e49a6c988fc63557a7a02df06ba02ff5b850f0d5a6d4ceR196-R201">`src/promptflow/tests/executor/unittests/_utils/test_multimedia_utils.py`</a>: Added test cases to improve test coverage for `load_multimedia_data` and `create_image` functions. <a href="diffhunk://#diff-032d36cff1825e9647e49a6c988fc63557a7a02df06ba02ff5b850f0d5a6d4ceR196-R201">[1]</a> <a href="diffhunk://#diff-032d36cff1825e9647e49a6c988fc63557a7a02df06ba02ff5b850f0d5a6d4ceL105-R105">[2]</a> <a href="diffhunk://#diff-032d36cff1825e9647e49a6c988fc63557a7a02df06ba02ff5b850f0d5a6d4ceR125-R133">[3]</a>
* <a href="diffhunk://#diff-46760b7ed265c4b29c9cc4111fb19da9b815717e4d7180f61e0d0da023d2a8f9R145-R148">`src/promptflow/promptflow/_utils/multimedia_utils.py`</a>: Updated exception handling for empty image inputs, including raising an `InvalidImageInput` exception with a specific error message, catching and re-raising the exception with a modified error message, and modifying the exception to include the input key in the error message. <a href="diffhunk://#diff-46760b7ed265c4b29c9cc4111fb19da9b815717e4d7180f61e0d0da023d2a8f9R145-R148">[1]</a> <a href="diffhunk://#diff-46760b7ed265c4b29c9cc4111fb19da9b815717e4d7180f61e0d0da023d2a8f9R228">[2]</a> <a href="diffhunk://#diff-46760b7ed265c4b29c9cc4111fb19da9b815717e4d7180f61e0d0da023d2a8f9R237-R240">[3]</a>

Improving error handling for invalid image inputs.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_utils/multimedia_utils.py [hunk 1]
```diff
@@ -221,14 +225,19 @@ def _process_recursively(value: Any, process_funcs: Dict[type, Callable] = None,
...
+                if isinstance(updated_inputs[key], list):
+                    # For aggregation node, the image input is a list.
+                    updated_inputs[key] = [create_image(item) for item in updated_inputs[key]]
+                else:
+                    updated_inputs[key] = create_image(updated_inputs[key])
+            elif value.type == ValueType.LIST or value.type == ValueType.OBJECT:
+                updated_inputs[key] = load_multimedia_data_recursively(updated_inputs[key])
+        except InvalidImageInput as ex:
+            raise InvalidImageInput(
+                message_format=f"Failed to load image for input '{key}': {ex.message}", target=ex.target
```
Comment (zzn2) [inline src/promptflow/promptflow/_utils/multimedia_utils.py:239]: We should not use the formatted value in `message_format`.

```suggestion
                message_format="Failed to load image for input '{key}': {error_type_and_message}",
                key=xxx,
                error_type_and_message=xxx,
                target=ex.target
```

File: src/promptflow/promptflow/_utils/multimedia_utils.py [hunk 2]
```diff
@@ -221,14 +225,19 @@ def _process_recursively(value: Any, process_funcs: Dict[type, Callable] = None,
...
+        try:
+            if value.type == ValueType.IMAGE:
+                if isinstance(updated_inputs[key], list):
+                    # For aggregation node, the image input is a list.
+                    updated_inputs[key] = [create_image(item) for item in updated_inputs[key]]
+                else:
+                    updated_inputs[key] = create_image(updated_inputs[key])
+            elif value.type == ValueType.LIST or value.type == ValueType.OBJECT:
+                updated_inputs[key] = load_multimedia_data_recursively(updated_inputs[key])
+        except InvalidImageInput as ex:
```
Comment (zzn2) [inline src/promptflow/promptflow/_utils/multimedia_utils.py:237]: Is it good to catch `InvalidImageInput` only? Should we catch more other exceptions?

**Review by zzn2 — DISMISSED** [review]

File: src/promptflow/promptflow/_utils/multimedia_utils.py (hunk 1 above)
Comment (lumoslnt) [reply src/promptflow/promptflow/_utils/multimedia_utils.py]: Updated.

File: src/promptflow/promptflow/_utils/multimedia_utils.py (hunk 2 above)
Comment (lumoslnt) [reply src/promptflow/promptflow/_utils/multimedia_utils.py]: Updated.

File: src/promptflow/promptflow/_utils/multimedia_utils.py [hunk 3]
```diff
@@ -263,3 +275,11 @@ def resolve_image_path(input_dir: Path, image_dict: dict):
             if resource == "path":
                 image_dict[key] = str(input_dir / image_dict[key])
     return image_dict
+
+
+class InvalidImageInput(ValidationException):
+    pass
+
+
+class LoadMultimediaDataError(UserErrorException):
```
Comment (zzn2) [inline src/promptflow/promptflow/_utils/multimedia_utils.py:284]: I think current convention is to define error codes in a centralized file?

Comment (lumoslnt) [reply src/promptflow/promptflow/_utils/multimedia_utils.py]: Defined the errors in the `_utils._errors.py`.

**Review by zzn2 — APPROVED** [review]

**Review by PeiwenGaoMS — APPROVED** [review]

**Pull request merged**